src/cplus_plugin/gui/constant_raster_widgets.py

```diff
@@ -0,0 +1,245 @@
+
+    def _create_manual_ui(self):
+        """Create UI manually if UI file doesn't exist."""
+        layout = QtWidgets.QVBoxLayout(self)
+
+        # Label
+        label = QtWidgets.QLabel("Years of Experience:")
+        layout.addWidget(label)
+
+        # Spin box
+        self.sb_experience = QtWidgets.QDoubleSpinBox()
+        self.sb_experience.setMinimum(0.0)
+        self.sb_experience.setMaximum(100.0)
+        self.sb_experience.setValue(0.0)
+        self.sb_experience.setDecimals(1)
+        layout.addWidget(self.sb_experience)
+
+        # Stretch
+        layout.addStretch()
+
+    def reset(self):
+        """Interface implementation."""
+        if hasattr(self, "sb_experience"):
+            self.sb_experience.blockSignals(True)
+            self.sb_experience.setValue(0.0)
+            self.sb_experience.blockSignals(False)
+
+    def load(self, raster_component: ConstantRasterComponent):
+        """Interface implementation."""
+        # Block signals temporarily to prevent notify_update from
+        # being called to save the value being loaded.
+        if hasattr(self, "sb_experience"):
+            self.sb_experience.blockSignals(True)
+            if raster_component and raster_component.value_info:
+                self.sb_experience.setValue(raster_component.value_info.absolute)
+            else:
+                self.sb_experience.setValue(0.0)
+            self.sb_experience.blockSignals(False)
+
+    def on_experience_value_changed(self, value: float):
+        """Slot raised when spin box value changes."""
+        if (
+            self._constant_raster_component
+            and self._constant_raster_component.value_info
+        ):
+            self._constant_raster_component.value_info.absolute = value
+        self.notify_update()
+
+    @classmethod
+    def create_raster_component(
+        cls, model_component: LayerModelComponent
+    ) -> ConstantRasterComponent:
+        """Interface implementation.
+
+        Creates a component with default value 0.0. The actual value
+        will be set later when the user enters it in the widget.
+        """
+        # Determine component type
+        component_type = ModelComponentType.UNKNOWN
+        if hasattr(model_component, "__class__"):
+            class_name = model_component.__class__.__name__
+            if class_name == "NcsPathway":
+                component_type = ModelComponentType.NCS_PATHWAY
+            elif class_name == "Activity":
+                component_type = ModelComponentType.ACTIVITY
+
+        return ConstantRasterComponent(
+            value_info=ConstantRasterInfo(absolute=0.0),  # Default value
+            component=model_component,
+            component_id=(
+                str(model_component.uuid) if hasattr(model_component, "uuid") else ""
+            ),
+            component_type=component_type,
+            skip_raster=False,
+        )
+
+    @classmethod
+    def create_metadata(
```

Comment:
@osundwajeff the new collection will be defined through the registry during plugin initialization then call `load` to get the last saved collection if it exists.